include/queue.h

```diff
@@ -67,6 +67,223 @@
     } \
   while (0)
 
+#define sq_addfirst(p, q) \
+  do \
+    { \
+      FAR sq_entry_t *tmp_node = p; \
+      tmp_node->flink = (q)->head; \
+      if (!(q)->head) \
+        { \
+          (q)->tail = tmp_node; \
+        } \
+      (q)->head = tmp_node; \
+    } \
+  while (0)
+
+#define dq_addfirst(p, q) \
+  do \
+    { \
+      FAR dq_entry_t *tmp_node = p; \
+      tmp_node->blink = NULL; \
+      tmp_node->flink = (q)->head; \
+      if (!(q)->head) \
+        { \
+          (q)->head = tmp_node; \
+          (q)->tail = tmp_node; \
+        } \
+      else \
+        { \
+          (q)->head->blink = tmp_node; \
+          (q)->head = tmp_node; \
+        } \
+    } \
+  while (0)
+
+#define sq_addlast(p, q) \
+  do \
+    { \
+      FAR sq_entry_t *tmp_node = p; \
+      tmp_node->flink = NULL; \
+      if (!(q)->head) \
+        { \
+          (q)->head = tmp_node; \
+          (q)->tail = tmp_node; \
+        } \
+      else \
+        { \
+          (q)->tail->flink = tmp_node; \
+          (q)->tail        = tmp_node; \
+        } \
+    } \
+  while (0)
+
+#define dq_addlast(p, q) \
+  do \
+    { \
+      FAR dq_entry_t *tmp_node = p; \
+      tmp_node->flink = NULL; \
+      tmp_node->blink = (q)->tail; \
+      if (!(q)->head) \
+        { \
+          (q)->head = tmp_node; \
+          (q)->tail = tmp_node; \
+        } \
+      else \
+        { \
+          (q)->tail->flink = tmp_node; \
+          (q)->tail        = tmp_node; \
+        } \
+    } \
+  while (0)
+
+#define dq_addbefore(n, p, q) \
+  do \
+    { \
+      FAR dq_entry_t *_tmp_node = p; \
```

Comment:
```suggestion
      FAR dq_entry_t *_tmp_node = (p); \
```
why we need `_` in `_tmp_node`?